modules/ocf_staffvm/manifests/init.pp

```diff
@@ -1,4 +1,9 @@
 class ocf_staffvm {
   include ocf_staffvm::docker_group
   include ocf_staffvm::firewall
+
+  $owner = lookup('owner', { 'default_value' => undef })
```

Comment:
I don't think you necessarily need to provide a default value here, if this lookup fails, then puppet will fail anyways:

> \<DEFAULT VALUE\> [...] Defaults to no value; if Hiera can’t find a normal value, the lookup (and compilation) will fail.

Having the customized error message does help to make this a bit nicer though, so that does make this approach better on that front.

Reply:
Yeah, I'm aware a lookup fails. But just putting a lookup in there that does not make it readily apparent what is going on.

Reply:
I mean you could put a comment explaining it if you want, and I think that would be pretty readable, but I think the real advantage of doing it the way you have here is the customizable error message, so let's just stick with this.

Reply:
@kkuehler If you want this custom error message you'll need to put the two `include` lines below the owner check. Otherwise Puppet will error out in the `lookup('owner')` statement in the `ocf_staffvm::docker_group` class instead.

Reply:
That's true, it'll only error if it's tagged to install docker, but it should still go before that include so that the custom error takes priority.